lib/rack/multipart/parser.rb

```diff
@@ -182,7 +182,7 @@ def initialize(boundary, tempfile, bufsize, query_parser)
         @collector = Collector.new tempfile
 
         @sbuf = StringScanner.new("".dup)
-        @body_regex = /(.*?)(#{EOL})?#{Regexp.quote(@boundary)}(#{EOL}|--)/m
+        @body_regex = /(?:#{EOL})?#{Regexp.quote(@boundary)}(?:#{EOL}|--)/m
```

Comment:
I tried that, but it does not seem to work, maybe because of the `rx_max_size` calculation one line below. EOL could be "\r\n" and `^` just matches "\n" so there is some difference.